### PR TITLE
#165413091 configure PostgreSQL as a db engine

### PR DESCRIPTION
**Pull request by Kabalisa**
#### Description
In our authors' haven project we are going to be working with a lot of data and that is why we need to configure a database to persist our data into. our database is going to be configured to use PostgreSQL as its DB engine and sequelize as an ORM.
#### Type of change
N/A
#### How Has This Been Tested?
* First, create two databases for the development environment and the test environment.
* run ```npm start``` to first migrate the table models available into your development database and then start the server.
* run ```npm run migrate``` to migrate available table models into your development database without starting the server.
* run ```npm run testmigration```   to migrate available table models into your test database without starting the server.
#### Checklist:
N/A
#### Pivotal Tracker ID
[#165413091](https://www.pivotaltracker.com/story/show/165413091)

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["@babel/preset-env"]
+}
```
Comment (Niyitangasam) [inline .babelrc]: You can add a new line here

File: .env [hunk 2]
```diff
@@ -0,0 +1,4 @@
+DB_USER=postgres
+DB_PASSWORD=Kif@0788475785
+DB_NAME=authorshaven
+TESTDB_NAME=authorshaventest
```
Comment (Niyitangasam) [inline .env]: You should have put .env in .gitignore

Comment (Niyitangasam) [reply .env]: Also, this code is not supposed to be pushed on github

File: .sequelizerc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require("@babel/register")
+const path = require('path');
+
+module.export = {
+    "config": path.resolve('config', 'config.js'),
+    "models-path": path.resolve('models'),
+    "migrations-path": path.resolve('migrations'),
+    "seeders-path": path.resolve('seeders')
+}
```
Comment (Niyitangasam) [inline .sequelizerc:9]: Please use ES6 syntax here

File: index.js [hunk 4]
```diff
@@ -1,93 +1,19 @@
...
 
 // finally, let's start our server...
-const server = app.listen(process.env.PORT || 3000, function() {
-    console.log("Listening on port " + server.address().port);
-});
+app.listen(port, () => {
+    console.log(`Listening on port ${port}`);
+  });
+
+export default app;
```
Comment (Niyitangasam) [inline index.js:11]: You can also rebase from develop branch because this index.js has changed

**Review by Niyitangasam — CHANGES_REQUESTED** [review]
You did a great Job, few things to change refer to  my inline comments.

File: config/config.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    },
+    production: {
+        username: process.env.DB_USER,
+        password: process.env.DB_PASSWORD,
+        database: process.env.DB_NAME,
+        host: "127.0.0.1",
+        port: 5432,
+        dialect: "postgres"
+    }
+};
```
Comment (Niyitangasam) [inline config/config.js:30]: Use ES6 here as well

**Comment by AJAkimana** [comment]
Great work Fiston. But remove .env file.

**Review by AJAkimana — CHANGES_REQUESTED** [review]
Remove .env file on repo

File: .sequelizerc (hunk 3 above)
Comment (macphilips) [reply .sequelizerc:9]: @Niyitangasam Sequeliize doesn't support es6

File: migrations/20190425110658-create-users.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        allowNull: false
+      },
+      email: {
+        type: Sequelize.STRING,
+        primaryKey: true,
+        allowNull: false
+      },
+      bio: {
+        type: Sequelize.STRING,
+        allowNull: false
```
Comment (macphilips) [inline migrations/20190425110658-create-users.js]: We need to allow null for users `bio` so we will be able to register a new user. When creating a user on registration event the information we are getting from the user doesn't include `bio`.

File: models/users.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+const users = (sequelize, DataTypes) => {
+  const users = sequelize.define('users', {
```
Comment (macphilips) [inline models/users.js]: change the model name, it has to be in the singular form.
```suggestion
  const users = sequelize.define('User', {
```

File: config/config.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+dotenv.config();
+
+module.exports = {
+    development: {
+        username: process.env.DB_USER,
+        password: process.env.DB_PASSWORD,
+        database: process.env.DB_NAME,
+        host: "127.0.0.1",
+        port: 5432,
+        dialect: "postgres"
```
Comment (macphilips) [inline config/config.js]: Make the dialect configurable
```suggestion
        dialect: process.env.DIALECT || "postgres"
```

File: config/config.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+module.exports = {
+    development: {
+        username: process.env.DB_USER,
```
Comment (macphilips) [inline config/config.js]: change this configuration to use a database URL string rather provide a username, password, and database name. It too bulky to provide all this information via an environment variable.

File: config/config.js (hunk 5 above)
Comment (macphilips) [inline config/config.js:30]: Also, you might want to consider refactoring this file as it filled with duplicate fields

File: package.json [hunk 10]
```diff
@@ -6,11 +6,20 @@
   "scripts": {
     "test": "nyc mocha --require @babel/register \"tests/**/*.js\"",
     "lint": "eslint --ignore-path .gitignore .",
-    "lint:fix": "npm run lint --fix"
+    "lint:fix": "npm run lint -- --fix",
+    "start": " ./node_modules/.bin/sequelize db:migrate && babel-node ./index.js",
```
Comment (macphilips) [inline package.json]:
```suggestion
    "start": " npm run migrate && babel-node ./index.js",
```

**Review by macphilips — CHANGES_REQUESTED** [review]
Nice work so far, but this is far from completion. I did not see any unit/integration test that tests the functionalities of your implementation. Also, attend to the comments I left below

File: .sequelizerc (hunk 3 above)
Comment (Niyitangasam) [reply .sequelizerc:9]: @macphilips Thanks, I wasn't aware.

**Review by Niyitangasam — APPROVED** [review]

**Review by Niyitangasam — CHANGES_REQUESTED** [review]
Great Job, but you can add samples of db config in .env.sample file.

**Review by Niyitangasam — APPROVED** [review]

**Review by AJAkimana — APPROVED** [review]

**Pull request merged**